### PR TITLE
Pointer adapter

### PR DESCRIPTION
**Pull request by gleblebedev**
A "Pointer adapter" that translates mouse, touch, keyboard and gamepad events into mouse event.

Allows to handle simple touch events as a mouse event.
Allows to move mouse cursor with gamepad on platforms that do not have mouse or touch support (game console for example)
Allows to move mouse cursor with keyboard on systems where mouse is not plugged in.

### CODE REVIEW

File: Source/Samples/117_PointerAdapter/PointerAdapterSample.cpp [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    // Create the UI content
+    CreateInstructions();
+
+    // Setup the viewport for displaying the scene
+    SetupViewport();
+    // Set the mouse mode to use in the sample
+    SetMouseMode(MM_FREE);
+    SetMouseVisible(true);
+
+    SubscribeToEvent(&pointerAdapter_, E_MOUSEMOVE, URHO3D_HANDLER(PointerAdapterSample, HandleMouseMove));
```
Comment (eugeneko) [inline Source/Samples/117_PointerAdapter/PointerAdapterSample.cpp:66]: Please don't use `URHO3D_HANDLER` in new code

File: Source/Urho3D/Input/PointerAdapter.cpp [hunk 2]
```diff
@@ -0,0 +1,371 @@
...
+    // Ignore mouse movement if touch is active.
+    if (activeTouchId_.has_value())
+        return;
+
+    using namespace MouseButtonDown;
+    UpdatePointer(pointerPosition_, true, false);
+}
+
+
+void PointerAdapter::HandleTouchBegin(StringHash eventType, VariantMap& args)
```
Comment (eugeneko) [inline Source/Urho3D/Input/PointerAdapter.cpp:202]: Let's avoid unused `StringHash eventType` in new code

File: Source/Urho3D/Input/PointerAdapter.h [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+#include "../Core/Object.h"
+#include "../Input/DirectionAggregator.h"
+
+namespace Urho3D
+{
+
+namespace PointerAdapterDetail
+{
+
+enum class SubscriptionMask : unsigned
```
Comment (eugeneko) [inline Source/Urho3D/Input/PointerAdapter.h:33]: General thought: why do we hide enum and expose bool flags, when we can just expose enum and *not* have 10 separate set/get functions?

Comment (gleblebedev) [reply Source/Urho3D/Input/PointerAdapter.h:33]: It feels like it suits Urho codebase better. If you prefer to change that we need to change other adapters too.

**Review by eugeneko — APPROVED** [review]

**Pull request merged**